PHP: Fails to compile when trying to compiling manually with `phpize`

Background: My goal here is to create a pre-compiled debian package for our [PHP runtime](https://github.com/googlecloudplatform/php-docker).

On our debian jessie based runtime, it fails to compile with the following commands:

```
# pecl download grpc
downloading grpc-1.0.1.tgz ...
Starting to download grpc-1.0.1.tgz (1,910,227 bytes)
.....................................................................................................done: 1,910,227 bytes
File //grpc-1.0.1.tgz downloaded
root@f2fc71a8f0a8:/# tar zxf grpc-1.0.1.tgz 
root@f2fc71a8f0a8:/# cd grpc-1.0.1
root@f2fc71a8f0a8:/grpc-1.0.1# phpize
Configuring for:
PHP Api Version:         20131106
Zend Module Api No:      20131226
Zend Extension Api No:   220131226
# ./configure
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for PHP prefix... /opt/php56
checking for PHP includes... -I/opt/php56/include/php -I/opt/php56/include/php/main -I/opt/php56/include/php/TSRM -I/opt/php56/include/php/Zend -I/opt/php56/inclu$e/php/ext -I/opt/php56/include/php/ext/date/lib
checking for PHP extension directory... /opt/php56/lib/x86_64-linux-gnu/extensions/no-debug-non-zts-20131226
checking for PHP installed headers prefix... /opt/php56/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... re2c
checking for re2c version... 0.13.5 (ok)
checking for gawk... no
checking for nawk... nawk
checking if nawk is broken... no
checking whether to enable grpc support... yes, shared
./configure: line 4107: cd: ../../grpc: No such file or directory
./configure: line 4138: cd: ../../grpc/src/php/ext: No such file or directory
./configure: line 4169: cd: ../../grpc/third_party/boringssl: No such file or directory
checking for ld used by cc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for /usr/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /usr/bin/nm -B
checking whether ln -s works... yes
checking how to recognize dependent libraries... pass_all
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking the maximum length of command line arguments... 1572864
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC
checking if cc PIC flag -fPIC works... yes
checking if cc static flag -static works... yes
checking if cc supports -c -o file.o... yes
checking whether the cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no

creating libtool
appending configuration tag "CXX" to libtool
configure: creating ./config.status
config.status: creating config.h
# make
...
 cc -Wall -Werror -Wno-parentheses-equality -Wno-unused-value -std=c11 -fvisibility=hidden -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN -D_HAS_EXCEPTIONS=0 -DNOMINMAX -I. -I/grpc-1.0.1 -DPHP_ATOM_INC -I/grpc-1.0.1/include -I/grpc-1.0.1/main -I/grpc-1.0.1 -I/opt/php56/include/php -I/opt/php56/include/php/main -I/opt/php56/include/php/TSRM -I/opt/php56/include/php/Zend -I/opt/php56/include/php/ext -I/opt/php56/include/php/ext/date/lib -I/include -I/grpc -DHAVE_CONFIG_H -g -O2 -c /grpc-1.0.1/src/core/lib/tsi/ssl_transport_security.c  -fPIC -DPIC -o src/core/lib/tsi/.libs/ssl_transport_security.o
/grpc-1.0.1/src/core/lib/tsi/ssl_transport_security.c: In function 'ssl_handshaker_extract_peer':
/grpc-1.0.1/src/core/lib/tsi/ssl_transport_security.c:967:3: error: implicit declaration of function 'SSL_get0_alpn_selected' [-Werror=implicit-function-declaration]
   SSL_get0_alpn_selected(impl->ssl, &alpn_selected, &alpn_selected_len);
   ^
/grpc-1.0.1/src/core/lib/tsi/ssl_transport_security.c: In function 'tsi_create_ssl_client_handshaker_factory':
/grpc-1.0.1/src/core/lib/tsi/ssl_transport_security.c:1366:7: error: implicit declaration of function 'SSL_CTX_set_alpn_protos' [-Werror=implicit-function-declaration]
       if (SSL_CTX_set_alpn_protos(
       ^
/grpc-1.0.1/src/core/lib/tsi/ssl_transport_security.c: In function 'tsi_create_ssl_server_handshaker_factory_ex':
/grpc-1.0.1/src/core/lib/tsi/ssl_transport_security.c:1521:7: error: implicit declaration of function 'SSL_CTX_set_alpn_select_cb' [-Werror=implicit-function-declaration]
       SSL_CTX_set_alpn_select_cb(impl->ssl_contexts[i],
       ^
/grpc-1.0.1/src/core/lib/tsi/ssl_transport_security.c: At top level:
cc1: error: unrecognized command line option "-Wno-parentheses-equality" [-Werror]
cc1: all warnings being treated as errors
Makefile:566: recipe for target 'src/core/lib/tsi/ssl_transport_security.lo' failed
make: *** [src/core/lib/tsi/ssl_transport_security.lo] Error 1

```

--- a/config.m4
+++ b/config.m4
@@ -5,9 +5,9 @@ if test "$PHP_GRPC" != "no"; then
   dnl Write more examples of tests here...
 
   dnl # --with-grpc -> add include path
-  PHP_ADD_INCLUDE(../../grpc/include)
-  PHP_ADD_INCLUDE(../../grpc/src/php/ext/grpc)
-  PHP_ADD_INCLUDE(../../grpc/third_party/boringssl/include)
+  PHP_ADD_INCLUDE(PHP_EXT_SRCDIR()/include)
+  PHP_ADD_INCLUDE(PHP_EXT_SRCDIR()/src/php/ext/grpc)
+  PHP_ADD_INCLUDE(PHP_EXT_SRCDIR()/third_party/boringssl/include)
 
   LIBS="-lpthread $LIBS"
 

--- a/templates/config.m4.template
+++ b/templates/config.m4.template
@@ -7,9 +7,9 @@
     dnl Write more examples of tests here...
 
     dnl # --with-grpc -> add include path
-    PHP_ADD_INCLUDE(../../grpc/include)
-    PHP_ADD_INCLUDE(../../grpc/src/php/ext/grpc)
-    PHP_ADD_INCLUDE(../../grpc/third_party/boringssl/include)
+    PHP_ADD_INCLUDE(PHP_EXT_SRCDIR()/include)
+    PHP_ADD_INCLUDE(PHP_EXT_SRCDIR()/src/php/ext/grpc)
+    PHP_ADD_INCLUDE(PHP_EXT_SRCDIR()/third_party/boringssl/include)
 
     LIBS="-lpthread $LIBS"
 

